### PR TITLE
Moved csvy parser to model reader

### LINKED ISSUE #1911: CSVY Parsers Should Be Incorporated to Model Readers
Currently, the CSVY parser functions are located with the parsers in tardis/io/parsers, but those functions belong in the tardis/io/model_reader.py file.  Please move all CSVY related parser functions to the model reader file.  Note that this will cause import statements to break and some tests to fail, so these things need to be fixed as well.

### PR DESCRIPTION
**Pull request by aman1971**
**Description**
Moved all function from `tardis/io/parsers/csvy.py` to `tardis/io/model_reader.py`

**Motivation and context**
- Related to GSOC project.
- Fixes #1911 

**How has this been tested?**
- [ ] Testing pipeline.
- [ x] Other.
      `Pytest`

**Type of change**

- [ ] Bug fix. 
- [ ] New feature. 
- [ x] Breaking change. 
       - Moved all function from `tardis/io/parsers/csvy.py` to `tardis/io/model_reader.py` and removed `csvy.py` 
       - and also removed the `tardis/io/tests/test_csvy_reader.py` and moved all `csvy_reader` related test function to ` 
          tardis/io/tests/test_model_reader.py`
- [ ] None of the above. 

**Checklist**

- [ ] My change requires a change to the documentation.
    - [ ] I have updated the documentation accordingly.
    - [ ] (optional) I have built the documentation on my fork following [the instructions](https://tardis-sn.github.io/tardis/contributing/development/documentation_guidelines.html#sharing-the-built-documentation-in-your-pr-documentation-preview).
- [ ] I have assigned and requested two reviewers for this pull request.

### CODE REVIEW

**Comment by marxwillia** [comment]
@aman1971 Great first PR!  Can you run Black (our PEP8 compliance checker) on the files that you changed?  Check out the docs here for more info:  https://tardis-sn.github.io/tardis/contributing/development/code_quality.html

**Comment by aman1971** [comment]
Thanks for the feedback @marxwillia. I have fixed the PEP8 compliance. Please have a test check run.

**Comment by andrewfullard** [comment]
It looks like black was run on a few too many files that are not related to the PR's purpose. This has led to merge conflicts. Can you revert the formatting changes and only run black on files you changed for this PR?

File: tardis/io/tests/test_model_reader.py [hunk 1]
```diff
@@ -123,3 +142,33 @@ def test_simple_read_cmfgen_density(cmfgen_fname):
...
+
+
+def test_csvy_nocsv_data_is_none(csvy_nocsv_fname):
+    yaml_dict, csv = load_csvy(csvy_nocsv_fname)
+    assert csv is None
+
+
+def test_missing_required_property(csvy_missing_fname):
+    yaml_dict, csv = load_csvy(csvy_missing_fname)
+    with pytest.raises(Exception):
```
Comment (Rodot-) [inline tardis/io/tests/test_model_reader.py:168]: Please be more specific about which `Exception` you are checking for.  Checking for a general Exception isn't really all that informative, and if an unexpected exception gets thrown here then the test will still pass.

File: tardis/conftest.py [hunk 2]
```diff
@@ -171,6 +171,7 @@ def config_montecarlo_1e5_verysimple():
     config = Configuration.from_yaml(path)
     return config
 
+
```
Comment (Rodot-) [inline tardis/conftest.py:174]: Please remove this extra space

**Pull request closed without merging**